cms/static_src/javascript/auth.js

```diff
@@ -0,0 +1,155 @@
+    return null;
+  }
+};
+
+// Initiates the logout process and handles redirection.
+const logout = async () => {
+  console.info('[WAGTAIL] Initiating logout...');
+  try {
+    const response = await fetchWithCsrf(logoutURL);
+    if (response?.ok) {
+      const redirectURL = response.redirected ? response.url : origin;
+      console.info('[WAGTAIL] Logout successful. Redirecting to:', redirectURL);
+      window.location.href = redirectURL;
+    } else {
+      console.error('[WAGTAIL] Logout failed:', response);
+      // Reload the page to handle any session inconsistencies.
+      window.location.reload();
+    }
+  } catch (error) {
+    console.error('[WAGTAIL] Exception during logout:', error);
+  }
+};
+
+// Session configuration
+const sessionConfig = {
+  timeOffsets: { passiveRenewal: parseInt(sessionRenewalOffsetSeconds, 10) * 1000 }, // milliseconds
+  apiEndpoints: {
+    renewSession: authTokenRefreshUrl,
+  },
+  onRenewSuccess: async (sessionExpiryTime, refreshExpiryTime) => {
+    console.debug(
+      `[WAGTAIL] Session renewed! (Session: ${sessionExpiryTime}, Refresh: ${refreshExpiryTime})`,
+    );
+    try {
+      const extendResponse = await fetchWithCsrf(extendSessionURL);
+      if (extendResponse?.ok) {
+        console.info('[WAGTAIL] Wagtail session extended successfully.');
+      } else {
+        console.error('[WAGTAIL] Session extension failed:', extendResponse);
+        await logout();
+      }
+    } catch (error) {
+      console.error('[WAGTAIL] Error extending session:', error);
+      await logout();
+    }
+  },
+  onRenewFailure: async (error) => {
+    console.error('[WAGTAIL] Session renewal failed:', error);
+    await logout();
+  },
+  onSessionValid: (sessionExpiryTime, refreshExpiryTime) => {
+    console.debug(
+      `[WAGTAIL] Session valid. (Session: ${sessionExpiryTime}, Refresh: ${refreshExpiryTime})`,
+    );
+  },
+  onSessionInvalid: async () => {
+    console.warn('[WAGTAIL] Session invalid.');
+    await logout();
+  },
+  onError: async (error) => {
+    console.error('[WAGTAIL] General error:', error);
+    await logout();
+  },
+};
+
+// Initialise SessionManagement only if the user is logged in via Cognito (indicated by the presence of an id_token cookie).
+// Don't load if it's in an iframe (e.g. the Wagtail admin interface preview).
+document.addEventListener('DOMContentLoaded', () => {
+  if (window.self === window.top && getCookieByName(idTokenCookieName)) {
+    SessionManagement.init(sessionConfig);
+
+    // These will be fetched from localStorage which is set by the auth service before redirect.
+    // If they do not exist, then the session is invalid and the user will be logged out.
+    // Only clear if neither value is set, i.e. brand-new tab with no prior expiry info:
+    if (
+      !localStorage.getItem('session_expiry_time') &&
+      !localStorage.getItem('refresh_expiry_time')
+    ) {
+      SessionManagement.setSessionExpiryTime(null, null);
+    }
```

Comment:
These values are never set by the auth client lib directly. Instead, they're stored in `dis_auth_client_state` key as:
```js
{
  session_expiry_time: "2025-06-04T09:12:39.000Z",
  refresh_expiry_time: "2025-06-04T20:57:15Z"
}
```
As a result, the current if condition always evaluates to false.

However, you don’t need this check at all. If you only call `setSessionExpiryTime` when `dis_auth_client_state` doesn’t exist, then the event listeners (used for session refresh, etc.) are never set up. This is because `setSessionExpiryTime` is the function that initialises those listeners.

Even when we pass `null` to `setSessionExpiryTime`, it still fetches the actual refresh timers from local storage.